File: 350. Intersection of Two Arrays II.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1, nums2) {
    if ( nums1 == null || nums2 == null ) {
        return null;
    }

    nums1.sort(function(a, b) {
        return a - b;
    });

    nums2.sort(function(a, b) {
        return a - b;
    });

    var i = 0;
    var j = 0;
    var index = 0;
    var result = [];

    while ( i < nums1.length && j < nums2.length ) {
        if ( nums1[i] == nums2[j] ) {
            result[index++] = nums1[i];
            i++;
            j++;
        } else if ( nums1[i] < nums2[j] ) {
            i++;
        } else {
            j++;
        }
    }

    return result;
};

File: 350. Intersection of Two Arrays II.py
class Solution(object):
    def intersect(self, nums1, nums2):
        result = []
        nums1 = sorted(nums1)
        nums2 = sorted(nums2)
        n1, n2 = 0, 0

        while n1 < len(nums1) and n2 < len(nums2):
            if nums1[n1] == nums2[n2]:
                result.append(nums1[n1])
                n1 += 1
                n2 += 1
            elif nums1[n1] < nums2[n2]:
                n1 += 1
            else:
                n2 += 1
        return result
